### PR TITLE
feature: log error message when error on server

### PR DESCRIPTION
**Pull request by TamarinEA**
https://jira.railsc.ru/browse/BPC-11892

### CODE REVIEW

File: lib/sphinx/integration/server_pool.rb [hunk 1]
```diff
@@ -36,14 +36,14 @@ def take_all
...
           begin
             yield server
-          rescue Exception
+          rescue Exception => e
             skip_servers << server
 
             if skip_servers.size >= servers.size
-              ::ThinkingSphinx.fatal("Error on servers: #{skip_servers.map(&:to_s).join(", ")}")
-              raise
+              ::ThinkingSphinx.fatal("Error on servers: #{skip_servers.map(&:to_s).join(", ")} - #{e.message}")
```
Comment (apress-hound) [inline lib/sphinx/integration/server_pool.rb]: Prefer single-quoted strings inside interpolations.

File: lib/sphinx/integration/server_pool.rb [hunk 2]
```diff
@@ -36,14 +36,14 @@ def take_all
         servers.each do |server|
           begin
             yield server
-          rescue Exception
+          rescue Exception => e
```
Comment (apress-hound) [inline lib/sphinx/integration/server_pool.rb]: Avoid rescuing the Exception class. Perhaps you meant to rescue StandardError?

File: lib/sphinx/integration/server_pool.rb [hunk 3]
```diff
@@ -15,15 +15,15 @@ def take
...
         begin
           yield server
-        rescue Exception
+        rescue Exception => e
           skip_servers << server
 
           if skip_servers.size >= @servers.size
-            ::ThinkingSphinx.fatal("Error on servers: #{skip_servers.map(&:to_s).join(", ")}")
-            raise
+            ::ThinkingSphinx.fatal("Error on servers: #{skip_servers.map(&:to_s).join(", ")} - #{e.message}")
```
Comment (apress-hound) [inline lib/sphinx/integration/server_pool.rb]: Prefer single-quoted strings inside interpolations.

File: lib/sphinx/integration/server_pool.rb [hunk 4]
```diff
@@ -15,15 +15,15 @@ def take
 
         begin
           yield server
-        rescue Exception
+        rescue Exception => e
```
Comment (apress-hound) [inline lib/sphinx/integration/server_pool.rb]: Avoid rescuing the Exception class. Perhaps you meant to rescue StandardError?

File: lib/sphinx/integration/server_pool.rb [hunk 5]
```diff
@@ -15,15 +15,15 @@ def take
...
           skip_servers << server
 
           if skip_servers.size >= @servers.size
-            ::ThinkingSphinx.fatal("Error on servers: #{skip_servers.map(&:to_s).join(", ")}")
-            raise
+            ::ThinkingSphinx.fatal("Error on servers: #{skip_servers.map(&:to_s).join(', ')} - #{e.message}")
+            raise e
```
Comment (c1n1c) [inline lib/sphinx/integration/server_pool.rb]: можно е не дописывать, и трейс еще залогировать

**Comment by isqad** [comment]
Увы, прав нет

**Comment by c1n1c** [comment]
теперь есть)

**Pull request merged**